### DISCUSSION TITLE
Can I make a plug-in to add Functionality to props?

### DISCUSSION THREAD

**Discussion by MechMel** [solidjs/solid#1965] [Q&A]
Is it possible to make a plug-in for Solid that adds extra features to props when they are parsed?

For example, I want to make prop assignment interface with our state API:
```tsx
function MyComponent(
  props: {
    myValue: Writable<number>;
  }
) {
  return (
    <Button onClick={() => (props.myValue = 42)}>Click Me</Button>
  );
}
```

Is it possible to make a plug-in that enables this? If so, where do I start?

**Comment by lxsmnsyc** [comment]
This is a bit tricky to do bottom-up, compiler-wise.

https://twitter.com/lxsmnsyc/status/1529283545746665472
